overlay-positioner.html

```diff
@@ -0,0 +1,173 @@
+     * should scroll if needed.
+     *
+     * @attribute sizingTarget
+     * @type Node
+     */
+
+    /**
+     * This object memoizes information needed to position and size the target element. By default
+     * the target element is centered along an axis if it is not positioned otherwise. Its size
+     * is constrained to the window size unless a max- or min-width/height is set.
+     *
+     * @attribute dimensions
+     * @type Object
+     */
+
+    get target() {
+      return this;
+    },
+
+    /**
+     * Memoize information needed to position and size the target element.
+     */
+    discoverDimensions: function() {
+      if (this.dimensions) {
+        return;
+      }
+      var target = getComputedStyle(this.target);
+      var sizer = getComputedStyle(this.sizingTarget);
+      this.dimensions = {
+        position: {
+          v: target.top !== 'auto' ? 'top' : (target.bottom !== 'auto' ?
+            'bottom' : null),
+          h: target.left !== 'auto' ? 'left' : (target.right !== 'auto' ?
+            'right' : null),
+          css: target.position
+        },
+        size: {
+          v: sizer.maxHeight !== 'none',
+          h: sizer.maxWidth !== 'none'
+        },
+        margin: {
+          top: parseInt(target.marginTop) || 0,
+          right: parseInt(target.marginRight) || 0,
+          bottom: parseInt(target.marginBottom) || 0,
+          left: parseInt(target.marginLeft) || 0
+        }
+      };
+    },
+
+    /**
+     * Resets the target element's position and size constraints, and clear
+     * the memoized data.
+     */
+    resetTargetDimensions: function() {
+      if (!this.dimensions || !this.dimensions.size.v) {
+        this.sizingTarget.style.maxHeight = '';
+        this.target.style.top = '';
+      }
+      if (!this.dimensions || !this.dimensions.size.h) {
+        this.sizingTarget.style.maxWidth = '';
+        this.target.style.left = '';
+      }
+      this.dimensions = null;
+    },
+
+    /**
+     * Constrain the size of the target element and position it if necessary.
+     */
+    updateTargetDimensions: function() {
+      this.sizeTarget();
+      this.repositionTarget();
+    },
+
+    /**
+     * Constrains the height and width of the target element if not already constrained.
+     */
+    sizeTarget: function() {
+      this.sizingTarget.style.boxSizing = 'border-box';
+      var dims = this.dimensions;
+      var rect = this.target.getBoundingClientRect();
+      if (!dims.size.v) {
+        this.sizeDimension(rect, dims.position.v, 'top', 'bottom', 'Height');
+      }
+      if (!dims.size.h) {
+        this.sizeDimension(rect, dims.position.h, 'left', 'right', 'Width');
+      }
+    },
+
+    /**
+     * Constrains one of the target element's dimensions to the window.
+     */
+    sizeDimension: function(rect, positionedBy, start, end, extent) {
+      var dims = this.dimensions;
+      var flip = (positionedBy === end);
+      var m = flip ? start : end;
```

Comment:
Regarding my comment about abbreviated identifiers above, this method is noticeably harder to grok.